dev/scripts/update-versions.sh

```diff
@@ -0,0 +1,119 @@
+#!/usr/bin/env bash
+#
+# The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+# (the "License"). You may not use this work except in compliance with the License, which is
+# available at www.apache.org/licenses/LICENSE-2.0
+#
+# This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+# either express or implied, as more fully set forth in the License.
+#
+# See the NOTICE file distributed with this work for information regarding copyright ownership.
+#
+
+# This script is designed so that someone can update all versions for releases
+# using a single command.
+
+# Debugging
+# set -x
+set -e # exit if a command fails
+set -u # unbound variable is an error
+
+HELP=$(cat <<EOF
+Usage: update-version.sh [-s] <old version> <new version>
+
+    <old version>    The version to replace throughout the codebase
+    <new version>    The new version in the codebase
+
+    -s               Skips updating maven pom versions. This should mainly be
+                     used when cutting a final release where pom versions are
+                     updated by the maven release plugin.
+
+
+EOF
+)
+
+# Arguments:
+#  $1: old version
+#  $2: new version
+function update_poms() {
+    find . -name pom.xml | xargs -t -n 1 perl -pi -e "s/${1}/${2}/g"
```

Comment:
is `perl` ok to assume? Is that more common than `sed`?

Reply:
As far as my experience has gone, I've found perl to be installed on pretty much every operating system that we develop on (Amazon linux, CentOS, Ubuntu, MacOS) and the interface and arguments that I am using with perl don't vary from distro to distro.

The reason I decided against `sed` is that the default `sed` installed on MacOS has a difference interface than GNU `sed` which is installed on the Linux-based OSes (Ubuntu, CentOS, Amazon Linux) when it comes to in-place editing with the `-i` argument.

[`-i` is not a part of the POSIX spec](https://pubs.opengroup.org/onlinepubs/9699919799/utilities/sed.html), so it seems the default used on MacOS doesn't behave the same as on the linux systems.

Reply:
Here are two additional threads I read before deciding to stick with perl

- https://unix.stackexchange.com/questions/92895/how-can-i-achieve-portability-with-sed-i-in-place-editing
- https://stackoverflow.com/questions/5694228/sed-in-place-flag-that-works-both-on-mac-bsd-and-linux